executor/aggfuncs/func_max_min.go

```diff
@@ -14,49 +14,138 @@
 package aggfuncs
 
 import (
+	"sort"
+
 	"github.com/pingcap/tidb/sessionctx"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/types/json"
 	"github.com/pingcap/tidb/util/chunk"
 	"github.com/pingcap/tidb/util/stringutil"
 )
 
+type maxMinQueue struct {
+	queue   []interface{}
+	counter map[interface{}]int64
```

Comment:
It seems that we can't apply `map[interface{}]...` on structs, here is my test:
```
type MyDecimal struct {
	digitsInt int8 // the number of *decimal* digits before the point.
}

func main() {
	m := make(map[interface{}]bool)
	d1 := &MyDecimal{digitsInt: 2}
	m[d1] = true
	d2 := &MyDecimal{digitsInt: 2}
	fmt.Println(m[*d2], (*d1 == *d2))
}
```
The result is: `false true`.

Reply:
how about using the struct directly like this

Reply:
I think I can use `ToHashKey` to generate the key of `MyDecimal`
```
	m := make(map[interface{}]bool)
	d1 := new(MyDecimal).FromInt(1)
	key1, _ := d1.ToHashKey()
	m[string(key1)] = true
	d2 := new(MyDecimal).FromInt(1)
	key2, _ := d2.ToHashKey()
	fmt.Println(m[string(key2)], (*d1 == *d2))
```
the result is `true true`

this implement avoids the memory copy.

Reply:
@qw4990

Reply:
I think it's okay.